app/helpers/submissions_helper.rb

```diff
@@ -44,25 +44,38 @@ def set_release_on_results(groupings, release)
   end
 
   def get_submissions_table_info(assignment, groupings)
-    parts = groupings.select &:has_submission?
-    results = Result.where(submission_id:
-                             parts.map(&:current_submission_used))
+    if current_user.is_a_reviewer?(assignment)
+      
+    else
+      parts = groupings.select &:has_submission?
+      results = Result.where(submission_id:
+                                 parts.map(&:current_submission_used))
                     .order(:id)
+    end
+
     groupings.map.with_index do |grouping, i|
       g = Hash.new
       begin # if anything raises an error, catch it and log in the object.
-        submission = grouping.current_submission_used
-        if submission.nil?
-          result = nil
-        elsif submission.submitted_remark.nil?
-          result = (results.select do |r|
-            r.submission_id == submission.id
-          end).first
+        if current_user.is_a_reviewer?(assignment)
+          # "groupings" are the reviewee groupings.
+          # Get the respective reviewee's result from grouping
+          result_pr = current_user.grouping_for(assignment.id).review_for(grouping)
+          result = Result.find(result_pr.result_id)
         else
-          result = submission.remark_result
+          submission = grouping.current_submission_used
+          if submission.nil?
+            result = nil
+          elsif submission.submitted_remark.nil?
+            result = (results.select do |r|
+              r.submission_id == submission.id
+            end).first
+          else
+            result = submission.remark_result
+          end
         end
+
         g[:name] = grouping.get_group_name
-        unless current_user.student?
+        unless (current_user.is_a?(Student) && !current_user.is_a_reviewer?(assignment))
```

Comment:
None or almost none of the information provided in this block is necessary for the reviews table.